pkg/steps/cluster_claim.go

```diff
@@ -45,81 +47,91 @@ func (s *clusterClaimStep) Validate() error {
 	return nil
 }
 
-func (s *clusterClaimStep) Run(ctx context.Context) error {
-	return results.ForReason("cluster_claim").ForError(s.run(ctx))
-}
-
-func (s *clusterClaimStep) Name() string {
-	return fmt.Sprintf("cluster-pool:%s:%s-%s-%s-%s-%s", s.as, s.clusterClaim.Product, s.clusterClaim.Version, s.clusterClaim.Architecture, s.clusterClaim.Cloud, s.clusterClaim.Owner)
-}
-
-func (s *clusterClaimStep) Description() string {
-	return fmt.Sprintf("Claim a(n) %s cluster of version %s and architecture %s on cloud %s with %s's account", s.clusterClaim.Product, s.clusterClaim.Version, s.clusterClaim.Architecture, s.clusterClaim.Cloud, s.clusterClaim.Owner)
-}
+func (s *clusterClaimStep) Name() string                        { return s.wrapped.Name() }
+func (s *clusterClaimStep) Description() string                 { return s.wrapped.Description() }
+func (s *clusterClaimStep) Requires() []api.StepLink            { return s.wrapped.Requires() }
+func (s *clusterClaimStep) Creates() []api.StepLink             { return s.wrapped.Creates() }
+func (s *clusterClaimStep) Objects() []ctrlruntimeclient.Object { return s.wrapped.Objects() }
+func (s *clusterClaimStep) Provides() api.ParameterMap          { return s.wrapped.Provides() }
 
-func (s *clusterClaimStep) Requires() []api.StepLink { return nil }
-
-func (s *clusterClaimStep) Creates() []api.StepLink {
-	return []api.StepLink{api.ClusterClaimLink(s.as)}
+func (s *clusterClaimStep) Run(ctx context.Context) error {
+	return results.ForReason("utilizing_cluster_claim").ForError(s.run(ctx))
 }
 
-func (s *clusterClaimStep) Provides() api.ParameterMap { return nil }
-
-func (s *clusterClaimStep) Objects() []ctrlruntimeclient.Object { return s.client.Objects() }
-
 func (s *clusterClaimStep) run(ctx context.Context) error {
 	if s.clusterClaim == nil {
 		// should never happen
 		return fmt.Errorf("cannot claim a nil cluster")
 	}
+	clusterClaim, err := acquireCluster(ctx, *s.clusterClaim, s.hiveClient, s.client, *s.jobSpec)
+	if err != nil {
+		return err
```

Comment:
@AlexNPavel can we get a follow-up to add a specific `results.ForReason` marker for this case?

Reply:
https://github.com/openshift/ci-tools/pull/1941